test/cpp/api/singleton_int.cpp

```diff
@@ -8,12 +8,17 @@
 #include <test/cpp/api/support.h>
 
 TEST(SingletonIntTest, Comparisons) {
-  auto a = c10::SymInt(
-      c10::SymNode(c10::make_intrusive<c10::SingletonSymNodeImpl>(1, 1)));
-  auto b = c10::SymInt(
-      c10::SymNode(c10::make_intrusive<c10::SingletonSymNodeImpl>(1, 1)));
-  auto c = c10::SymInt(
-      c10::SymNode(c10::make_intrusive<c10::SingletonSymNodeImpl>(2, 1)));
+  auto x = torch::randn({2, 2});
+
+  auto a =
+      c10::SymInt(c10::SymNode(c10::make_intrusive<c10::SingletonSymNodeImpl>(
+          1, 1, x, 1, c10::SingletonVariant::PYTHON)));
+  auto b =
+      c10::SymInt(c10::SymNode(c10::make_intrusive<c10::SingletonSymNodeImpl>(
+          1, 1, x, 1, c10::SingletonVariant::PYTHON)));
+  auto c =
+      c10::SymInt(c10::SymNode(c10::make_intrusive<c10::SingletonSymNodeImpl>(
+          2, 1, x, 1, c10::SingletonVariant::PYTHON)));
```

Comment:
How exactly do the C++ only tests work now:? 🤔

Reply:
Yeah the naming is kind of unfortunate, but we're using `PYTHON` variant NTs in C++ because only python variant NTs support comparisons.

I named it `PYTHON` because it attaches the Python key, but open to any other ideas as well

Reply:
So you have to carefully avoid hitting dispatcher in these tests? Good to doc.